docs/explanation/filesets.rst

```diff
@@ -0,0 +1,107 @@
+.. _filesets_explanation:
+
+Filesets
+========
+
+Filesets are named collections of files and directories that can be migrated
+between steps in the process of :ref:`building a part <parts>`. They are used
+within Craft Parts to collect and filter the files and directories a part
+needs in its *stage* and *prime* processing :ref:`steps <lifecycle>`.
+
+Tools that depend on Craft Parts can use filesets to simplify and automate
+migration of files and directories between build steps. Users of those tools
+may need to know about filesets if they need to adjust the contents of the
+packages that the tools produce.
+
+Fileset names
+-------------
+
+Internally, Craft Parts uses the ``overlay``, ``stage`` and ``prime`` filesets
+to migrate files from all parts into the corresponding steps. For example,
+the *stage* fileset refers to all the files and directories that will be moved
+into the staging area ready for the *stage* step to be run.
+
+Tools that use Craft Parts can define filesets with arbitrary names and they
+can also expose the feature to users. Names are only limited by the format
```

Comment:
> Names are only limited by the format used to describe them.

This took me a minute to comprehend.  Is this saying that there are no naming conventions or restrictions for filesets other than the restrictions of the config file format?  (as in yaml's syntax rules when filesets are defined in a snapcraft.yaml file?)

Pedantic: I would reword this to be more to-the-point.  Something like "there are no restrictions on fileset names."  Your statement is true, but it's true for any data written in any format; it's distracting in this context.